src/app/features/home/services/card.service.ts

```diff
@@ -1,20 +1,28 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import { Observable } from 'rxjs';
+import { map, Observable } from 'rxjs';
 import { environment } from "../../../../environments/environment";
-import {ICard} from "../../../core/models/cards";
-
-
+import { ICard } from "../../../core/models/cards";
 
 @Injectable({
   providedIn: 'root'
 })
 export class CardService {
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient) {
+  }
 
-  getCards(cardUrl:string): Observable<Array<ICard>> {
+  getCards(cardUrl: string): Observable<Array<ICard>> {
     return this.http.get(environment.baseUrl + cardUrl) as Observable<Array<ICard>>
   }
 
+  getCardById(cardId: string): Observable<ICard | undefined> {
+    if(cardId.includes('s')) {
+      return this.http.get<ICard[]>(environment.baseUrl + 'sellCards.json')
+        .pipe(map(data => data.find(({id}) => cardId === id)))
+    } else {
+      return this.http.get<ICard[]>(environment.baseUrl + 'rentCards.json')
+        .pipe(map(data => data.find(({id}) => cardId === id)))
+    }
```

Comment:
This logic looks weird
You need 2 separate methods: getSellCardById and getRentCardById. A component should decide what method should be called. Common handler (pipe(map(data => data.find(({id}) => cardId === id)))) could be extracted to separate function for avoiding duplication